### ISSUE TITLE
Cannot train with raw dataset

### ISSUE THREAD

**Issue by kisenera** [NVlabs/stylegan2-ada#40]
Hello, when I try to train using a raw dataset (even when using --use-raw=True), it gives the following errors (for every config, as well):

```
Traceback (most recent call last):
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\client\session.py", line 1356, in _do_call
    return fn(*args)
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\client\session.py", line 1341, in _run_fn
    options, feed_dict, fetch_list, target_list, run_metadata)
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\client\session.py", line 1429, in _call_tf_sessionrun
    run_metadata)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Cannot batch tensors with different shapes in component 0. First element had shape [3,512,512] and element 2 had shape [1,512,512].
         [[{{node Dataset_1/IteratorGetNext}}]]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "train.py", line 620, in <module>
    main()
  File "train.py", line 612, in main
    run_training(**vars(args))
  File "train.py", line 501, in run_training
    training_loop.training_loop(**training_options)
  File "F:\Downloads 2\stylegan2-ada-main\training\training_loop.py", line 135, in training_loop
    grid_size, grid_reals, grid_labels = setup_snapshot_image_grid(training_set)
  File "F:\Downloads 2\stylegan2-ada-main\training\training_loop.py", line 33, in setup_snapshot_image_grid
    reals, labels = training_set.get_minibatch_np(gw * gh)
  File "F:\Downloads 2\stylegan2-ada-main\training\dataset.py", line 182, in get_minibatch_np
    return tflib.run(self._tf_minibatch_np)
  File "F:\Downloads 2\stylegan2-ada-main\dnnlib\tflib\tfutil.py", line 33, in run
    return tf.get_default_session().run(*args, **kwargs)
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\client\session.py", line 950, in run
    run_metadata_ptr)
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\client\session.py", line 1173, in _run
    feed_dict_tensor, options, run_metadata)
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\client\session.py", line 1350, in _do_run
    run_metadata)
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\client\session.py", line 1370, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Cannot batch tensors with different shapes in component 0. First element had shape [3,512,512] and element 2 had shape [1,512,512].
         [[node Dataset_1/IteratorGetNext (defined at F:\Downloads 2\stylegan2-ada-main\training\dataset.py:165) ]]

Errors may have originated from an input operation.
Input Source operations connected to node Dataset_1/IteratorGetNext:
 Dataset/IteratorV2 (defined at F:\Downloads 2\stylegan2-ada-main\training\dataset.py:147)

Original stack trace for 'Dataset_1/IteratorGetNext':
  File "train.py", line 620, in <module>
    main()
  File "train.py", line 612, in main
    run_training(**vars(args))
  File "train.py", line 501, in run_training
    training_loop.training_loop(**training_options)
  File "F:\Downloads 2\stylegan2-ada-main\training\training_loop.py", line 135, in training_loop
    grid_size, grid_reals, grid_labels = setup_snapshot_image_grid(training_set)
  File "F:\Downloads 2\stylegan2-ada-main\training\training_loop.py", line 33, in setup_snapshot_image_grid
    reals, labels = training_set.get_minibatch_np(gw * gh)
  File "F:\Downloads 2\stylegan2-ada-main\training\dataset.py", line 180, in get_minibatch_np
    self._tf_minibatch_np = self.get_minibatch_tf()
  File "F:\Downloads 2\stylegan2-ada-main\training\dataset.py", line 165, in get_minibatch_tf
    images, labels = self._tf_iterator.get_next()
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\data\ops\iterator_ops.py", line 426, in get_next
    output_shapes=self._structure._flat_shapes, name=name)
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\ops\gen_dataset_ops.py", line 1974, in iterator_get_next
    output_shapes=output_shapes, name=name)
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 788, in _apply_op_helper
    op_def=op_def)
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\util\deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\framework\ops.py", line 3616, in create_op
    op_def=op_def)
  File "F:\zPL Folder\a\lib\site-packages\tensorflow\python\framework\ops.py", line 2005, in __init__
    self._traceback = tf_stack.extract_stack()
```
Thanks in advance for any help

**Comment by kisenera** [comment]
Actually, this error doesn't seem to be exclusive to raw datasets. It seems to happen randomly for normal ones too. However, with a combination of certain config and tfrecords, I was able to get normal dataset to begin training, but never got a raw one to work.
It also may be worth noting. I have a 6GB GPU. Pretty weak when compared to the recommended, but I was able to train fine on the normal stylegan2.
